.github/workflows/nomad-requirements.yml

```diff
@@ -0,0 +1,56 @@
+name: Validate Dependencies Compatibility
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+    # Run workflow only when there are changes in pyproject.toml or dev-requirements.txt
+    # paths:
+    # - 'pyproject.toml'
+    # - 'dev-requirements.txt'
+
+jobs:
+  validate_dependencies:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout pynxtools
+      uses: actions/checkout@v2
+
+    - name: Checkout NOMAD from GitLab
+      run: |
+        git clone https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-FAIR.git nomad
```

Comment:
You could also add `--depth 1` here, that way the ci/cd would only checkout nomad shallowly (faster).

Reply:
Also you could use `--branch develop` here and `--recurse-submodules` then you don't have to add the lines below

Reply:
changed